File: examples/leaderboard.mjs
// Leaderboard CSV download (Node.js)
// Usage: node leaderboard.mjs [type] [year] [player_type]
const TYPE = process.argv[2] || "expected_statistics";
const YEAR = process.argv[3] || "2025";
const PLAYER_TYPE = process.argv[4] || "batter";

const url = `https://baseballsavant.mlb.com/leaderboard/${TYPE}?year=${YEAR}&type=${PLAYER_TYPE}&min=q&csv=true`;

console.log(`Downloading ${TYPE} leaderboard (${YEAR}, ${PLAYER_TYPE})...`);
const res = await fetch(url);
if (!res.ok) throw new Error(`Request failed: HTTP ${res.status}`);
const csv = await res.text();

const header = csv.split("\n")[0].split(",");
console.log(`Columns (${header.length}):`);
console.log(header.join(", "));
console.log(`Rows: ${csv.split("\n").filter(Boolean).length - 1}`);

File: examples/leaderboard.py
#!/usr/bin/env python3
"""Leaderboard CSV download (Python).

Usage: python leaderboard.py [type] [year] [player_type]
Requires: requests
"""
import sys

import requests

TYPE = sys.argv[1] if len(sys.argv) > 1 else "expected_statistics"
YEAR = sys.argv[2] if len(sys.argv) > 2 else "2025"
PLAYER_TYPE = sys.argv[3] if len(sys.argv) > 3 else "batter"

url = (
    f"https://baseballsavant.mlb.com/leaderboard/{TYPE}"
    f"?year={YEAR}&type={PLAYER_TYPE}&min=q&csv=true"
)

print(f"Downloading {TYPE} leaderboard ({YEAR}, {PLAYER_TYPE})...")
r = requests.get(url, timeout=60)
r.raise_for_status()

lines = r.text.splitlines()
header = lines[0].split(",") if lines else []
print(f"Columns ({len(header)}): {', '.join(header)}")
print(f"Rows: {max(len(lines) - 1, 0)}")